### PR TITLE
Automate check that changes in the profile of the user (data) happen

### PR DESCRIPTION
**Pull request by amurKontur**
## Release Notes

- **New Features**
	- Enhanced test cases for OpenStreetMap functionality, including improved navigation and profile updates.
	- Added methods for setting user profile details, enhancing user experience when updating information.

- **Bug Fixes**
	- Improved synchronization in test cases for better control flow and efficiency.

- **Chores**
	- Updated dependencies to the latest versions for better performance and stability.
	- Modified Playwright configuration to focus on relevant test scenarios.

### CODE REVIEW

File: e2e/openOsmUpdateProfileWithPro.spec.ts [hunk 1]
```diff
@@ -39,34 +40,56 @@ for (const project of projects) {
...
-test.describe(`As PRO User, I can use different OSM editors to open the map`, () => {
+test.describe(`As PRO User, I can use different OSM editors to open the map and I can update details of my profile`, () => {
   test.describe.configure({ mode: 'serial' });
   for (const project of limitedProjects) {
     for (const [editor, pattern] of editors) {
-      test(`As PRO User, I can go to ${project.title}, open map and open ${editor} at map coordinates`, async ({
+      test(`As PRO User, I can go to ${project.title}, update profile details, open map and open ${editor} at map coordinates`, async ({
         context,
         pageManager,
+        page,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmUpdateProfileWithPro.spec.ts:50]: well, I know, but it is better from the side of maintainability and speed of test run. Plus I have test names renamed to show real picture what is checked

File: e2e/openOsmUpdateProfileWithPro.spec.ts [hunk 2]
```diff
@@ -39,34 +40,53 @@
...
+        await pageManager.atProfilePage.setBioValue(newBio);
+        await pageManager.atProfilePage.setFullNameValue(newFullName);
         if (osmEditorValue !== editor) {
           await pageManager.atProfilePage.setOsmEditorValue(editor);
+          await pageManager.atProfilePage.saveChanges();
           const editedOsmEditorValue =
             await pageManager.atProfilePage.getOsmEditorValue();
           expect(editedOsmEditorValue).toEqual(editor);
+        } else {
+          await pageManager.atProfilePage.saveChanges();
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmUpdateProfileWithPro.spec.ts:66]: it is done by playwright framework itself

File: e2e/openOsmUpdateProfileWithPro.spec.ts [hunk 3]
```diff
@@ -39,34 +40,56 @@
...
         }
+        const newProfileData = await pageManager.atProfilePage.getProfileData(project, {
+          shouldOsmEditorBeSeenOnAtlas: true,
+        });
+        expect(newProfileData.bioValue, 'Bio should be updated').toStrictEqual(newBio);
+        expect(newProfileData.fullNameValue, 'Full name should be updated').toStrictEqual(
+          newFullName,
+        );
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmUpdateProfileWithPro.spec.ts:74]: no, we don't need retries here

File: e2e/openOsmWithUser.spec.ts [hunk 4]
```diff
@@ -11,18 +11,23 @@
...
   }) => {
     await pageManager.atBrowser.openProject(project, { skipCookieBanner: true });
-    await pageManager.atNavigationMenu.clickButtonToOpenPage('Map');
-
-    // TO DO: remove this action after 18582 issue is fixed
-    await pageManager.atMap.goToSpecificAreaByUrl(10.597, 53.9196, 27.5097, project);
+    await Promise.all([
+      pageManager.atNavigationMenu.clickButtonToOpenPage('Map'),
+      pageManager.atMap.waitForUrlToMatchPattern(/\?map=/i, page, 'domcontentloaded'),
+    ]);
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmWithUser.spec.ts:20]: no we should wait right for dom here

File: e2e/openOsmWithUser.spec.ts [hunk 5]
```diff
@@ -11,18 +11,26 @@ for (const project of projects) {
...
 
+    const editMapBtn = pageManager.atToolBar.getButtonByText('Edit map in OSM', page);
+    await editMapBtn.hover();
+
     const [newPage] = await Promise.all([
-      context.waitForEvent('page'),
-      (await pageManager.atToolBar.getButtonByText('Edit map in OSM')).click({
+      context.waitForEvent('page', { timeout: 25000 }),
+      editMapBtn.click({
         delay: 330,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmWithUser.spec.ts:34]: we should rely on text here

File: e2e/page-objects/helperBase.ts [hunk 6]
```diff
@@ -152,6 +152,15 @@ export class HelperBase {
...
   }
 
+  /**
+   * This method waits 12 secs, designed for waiting for zoom on map.
+   */
+
+  async waitForZoom() {
+    const zoomTimeout = 12000;
+    await this.page.waitForTimeout(zoomTimeout);
+  }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/helperBase.ts:162]: I know, but we can not wait for anything here. Not sure about MAP. timeout is not configurable and setting a variable just to set a variable does not make sense

File: e2e/page-objects/profilePage.ts [hunk 7]
```diff
@@ -94,6 +94,36 @@
...
+
+  /**
+   *  This method sets a new value for the bio field.
+   * @param newBio - new bio in string format
+   */
+
+  async setBioValue(newBio: string) {
+    const bioLocator = this.page.getByText('Bio').locator('..').locator('textarea');
+    await bioLocator.fill(newBio);
+  }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/profilePage.ts:120]: getting Bio by text firstly adds an extra check
if selector is not found, we can see an issue

**Review by dqunbp — APPROVED** [review]

File: e2e/openOsmWithUser.spec.ts [hunk 8]
```diff
@@ -11,18 +11,23 @@ for (const project of projects) {
...
 
+    const editMapBtn = pageManager.atToolBar.getButtonByText('Edit map in OSM');
+    await editMapBtn.hover();
+
     const [newPage] = await Promise.all([
       context.waitForEvent('page'),
-      (await pageManager.atToolBar.getButtonByText('Edit map in OSM')).click({
+      editMapBtn.click({
         delay: 330,
       }),
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsmWithUser.spec.ts]: delay here is to emulate real user click

File: e2e/openOsm.spec.ts [hunk 9]
```diff
@@ -17,19 +17,22 @@ for (const project of projects) {
...
+      const editMapBtn = pageManager.atToolBar.getButtonByText('Edit map in OSM');
+      await editMapBtn.hover();
 
       const [newPage] = await Promise.all([
         context.waitForEvent('page'),
-        (await pageManager.atToolBar.getButtonByText('Edit map in OSM')).click({
-          delay: 150,
+        editMapBtn.click({
+          delay: 330,
         }),
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/openOsm.spec.ts]: hover has waitFor included

**Pull request merged**